client/opkclient_test.go

```diff
@@ -103,43 +103,31 @@ func TestClient(t *testing.T) {
 				t.Fatalf("expected jkt header to be a string, got %T", jkt)
 			}
 
-			pubkey, err := op.PublicKey(context.Background(), nil)
+			opToken, err := pkt.Compact(pkt.Op)
 			require.NoError(t, err, tc.name)
 
-			pub, err := jwk.FromRaw(pubkey)
+			pubkey, err := provider.Verifier().ProviderPublicKey(context.Background(), opToken)
+			if err != nil {
+				t.Fatal(err)
+			}
 			require.NoError(t, err, tc.name)
 
-			thumbprint, err := pub.Thumbprint(crypto.SHA256)
+			thumbprint, err := pubkey.Thumbprint(crypto.SHA256)
 			require.NoError(t, err, tc.name)
 
 			thumbprintStr := string(util.Base64EncodeForJWT(thumbprint))
 			require.Equal(t, jktstr, thumbprintStr, "jkt header does not match op thumbprint in "+tc.name)
 
-			alg, ok := pkt.ProviderAlgorithm()
+			providerKeyAlgorithm, ok := pkt.ProviderAlgorithm()
```

Comment:
There are many instances in test files where we have to use different names besides `alg` because there are multiple variables to which that name would apply. That is also the case here. I was trying to complement your naming choice in `signerAlg`. I can shorten it to `providerAlg` which I think would make sense, but since you don't suggest that, what do you suggest I do?

Reply:
Either `providerAlg` or `opAlg` works for me. it is improvement in readability.

The function `ProviderAlgorithm` on the PK Token confusing because we have OpenIDProviders and CosignerProviders, but given that that the name `ProviderAlgorithm` is not a change in this PR, I would not advocate for a rename. We might just want to move away from calling the CosignerProvider a provider.